### ISSUE TITLE
Post here for missing commands

### ISSUE THREAD

**Issue by bezmi** [bezmi/homeassistant_jvc_projector_remote#17]
Newer models will undoubtedly add newer picture modes, commands and other settings. Post here if something for your projector is missing.

To get this one started

**NZ/RS4100/RS3100 (2021) Series Projectors**
- [ ]  add 'mid' lamp power mode

**Comment by ziptbm** [comment]
Thank you, @bezmi - Yes, it was HDR content using 'Frame Adapt HDR' that presented the error I shared from [issue 16](https://github.com/bezmi/homeassistant_jvc_projector_remote/issues/16).   I have an RS-1000.

**Comment by bezmi** [comment]
I have pushed a new alpha release (v2.0.0a3) which should fix your issues. Please try it and report back, I am inclined to make this a full v2.0.0 release when your issues have been worked out.

**Comment by roninf** [comment]
Hi @bezmi  im not sure if i am doing something or if it is not implemented yet.
But how can i choose the picture mode "frame_adapt_hdr2" and "frame_adapt_hdr3"? (NZ Series)
The HA remote shows "18" and "19" as mode when chosen via the physical remote but you cannot switch to these via the integration.
Can this be done ?

**Comment by blackgold9** [comment]
The latest NZ firmware supports laser dimming off/1/2/3. Would love to see support for that in the library. Thanks for your work!

**Comment by bezmi** [comment]
Thanks for the requests @roninf and @blackgold9. They should be easy to implement once I find the command codes. I'll push a minor update once I've added them.

**Comment by thoechsmann** [comment]
My NZ7 does not have mask-custom1-3. It can only enable and disable mask. 

so command mask-off works, but I cannot enable it again.

**Comment by bezmi** [comment]
@thoechsmann so you are requesting a mask enable implementation? If anyone has the updated command specification pdf, that would be very helpful.

**Comment by thoechsmann** [comment]
Yes, but looking at the latest available spec I found online for the remote commands there is no on state, just custom 1,2,3. Any NZ user out there that uses the mask custom1/2/3 successfully to enable masking?

**Comment by bezmi** [comment]
I have found a spec for the 2018 models, which also only appear to have on and off. According to that spec, our `custom2` command should send the required value for `on`. Could you please try sending the command `mask-custom2` via a `remote.send_command`  service call and let me know if that works?

**Comment by shiscarp** [comment]
Hi! Do you have a command for reading the lens memory state?

**Comment by Sawtaytoes** [comment]
I'm also wondering if you can read the current lens memory state.

**Comment by bezmi** [comment]
Sorry guys, the projector I have access to is outdated compared to yours, which makes implementing/testing commands very difficult. @thoechsmann have you got a link for the spec you found?
